.github/workflows/build.yml

```diff
@@ -42,46 +42,59 @@ jobs:
         python-version: ${{ matrix.python-version }}
         architecture: 'x64'
     - name: Set up Node
-      uses: actions/setup-node@v1
+      uses: actions/setup-node@v2
       with:
-         node-version: '12.18'
-    - name: Create npm configuration for authentication
-      run: |
-        echo "scripts-prepend-node-path=true" >> ~/.npmrc
-        echo "prefix=~/.npm-global" >> ~/.npmrc
-        mkdir -p ~/.npm-global
+         node-version: '12.x'
     - name: Install Python dependencies
       run: |
         python -m pip install --upgrade pip wheel jupyter_server
         python -m pip install --upgrade jupyterlab
-    - name: Install NPM dependencies
-      run: |
-        sudo apt install yarn
-        npm install -g lerna@3.22.1
-        npm install -g rimraf@3.0.2
-        npm install -g typescript@4.1.3
-        npm install -g yarn@1.22.4
     - name: Log current build tools version
       run: |
         python --version
         node --version
         npm --version
         yarn --version
-        lerna --version
         pip --version
     - name: Log current Python dependencies version
       run: |
         pip freeze
-    - name: Build
+    - name: Install dependencies
       run: |
-        export PATH=~/.npm-global/bin:$PATH
-        make install
+        yarn install --frozen-lockfile
+        pip install -q -r test_requirements.txt
+        python setup.py bdist_wheel sdist
+        pip install --upgrade --upgrade-strategy only-if-needed --use-deprecated=legacy-resolver dist/elyra-*-py3-none-any.whl
+        yarn lerna run build
     - name: Documentation
       run: |
         make docs
-    - name: Test
+    - name: Lint
+      run: |
+        yarn eslint:check
+        yarn prettier:check
+        flake8 elyra
+    - name: Unit tests
+      run: |
+        pytest -v elyra
+        yarn test:unit
+    - name: Integration tests
       run: |
-        make test
+        yarn lerna exec "yarn pack --filename dist.tgz"
+        yarn lerna clean
+        rm -rf node_modules
+        jupyter labextension install \
+          packages/code-snippet/dist.tgz \
+          packages/metadata/dist.tgz \
+          packages/metadata-common/dist.tgz \
+          packages/pipeline-editor/dist.tgz \
+          packages/python-editor/dist.tgz \
+          packages/r-editor/dist.tgz \
+          packages/script-editor/dist.tgz \
+          packages/services/dist.tgz \
+          packages/theme/dist.tgz \
+          packages/ui-components/dist.tgz
```

Comment:
Multiple things here, most of these should be part of make, `rm -rf node_modules` should be part of clean, the `lab extensions` some are to be linked others to be installed. Having in make as we have today avoid having this duplications that can introduce errors.